src/main/scala/za/co/absa/hyperdrive/trigger/persistance/WorkflowRepository.scala

```
@@ -109,43 +123,53 @@ class WorkflowRepositoryImpl extends WorkflowRepository {
     val deleteEvent = eventTable.filter(_.sensorId in deleteSensor.map(_.id))
     val deleteDagIns = dagInstanceTable.filter(_.workflowId === id)
     val deleteDagDef = dagDefinitionTable.filter(_.workflowId === id)
     val deleteJobDef = jobDefinitionTable.filter(_.dagDefinitionId in deleteDagDef.map(_.id))
     val deleteJobIns = jobInstanceTable.filter(_.dagInstanceId in deleteDagIns.map(_.id))
     val deleteWorkflow = workflowTable.filter(_.id === id)
-    db.run(
+
+    db.run(getWorkflowJoined(id).flatMap(
+      workflow => {
+        workflowHistoryRepository.delete(workflow, user)
+      }
+    ).flatMap(_ =>
       deleteEvent.delete andThen
         deleteSensor.delete andThen
         deleteJobDef.delete andThen
         deleteJobIns.delete andThen
         deleteDagIns.delete andThen
         deleteDagDef.delete andThen
         deleteWorkflow.delete andThen
-        DBIO.successful((): Unit).transactionally
-    )
+        DBIO.successful((): Unit
+    )).transactionally)
   }
 
-  override def updateWorkflow(workflow: WorkflowJoined)(implicit ec: ExecutionContext): Future[Either[ApiError, Unit]] = {
+  override def updateWorkflow(workflow: WorkflowJoined, user: String)(implicit ec: ExecutionContext): Future[Either[ApiError, Unit]] = {
     db.run((for {
       w <- workflowTable.filter(_.id === workflow.id).update(workflow.toWorkflow.copy(updated = Option(LocalDateTime.now())))
       s <- sensorTable.filter(_.workflowId === workflow.id).update(workflow.sensor)
       dd <- dagDefinitionTable.filter(_.workflowId === workflow.id).update(workflow.dagDefinitionJoined.toDag())
       deleteJds <- jobDefinitionTable.filter(_.dagDefinitionId === workflow.dagDefinitionJoined.id).delete
       insertJds <- jobDefinitionTable ++= workflow.dagDefinitionJoined.jobDefinitions.map(_.copy(dagDefinitionId = workflow.dagDefinitionJoined.id))
-    } yield {}).transactionally.asTry
-    ).map {
+    } yield {
+      w
+    }).asTry.map {
```

I think the `.asTry.map` section should be the last one. If there's an error while inserting the history entry, the error will be swallowed silently.

---

Same in `switchWorkflowActiveState`

---

Fixed